thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/detection/yaml/YamlDetectionConfigTranslatorTest.java

```diff
@@ -0,0 +1,46 @@
+package com.linkedin.thirdeye.detection.yaml;
+
+import com.linkedin.thirdeye.datalayer.bao.DAOTestBase;
+import com.linkedin.thirdeye.datalayer.bao.MetricConfigManager;
+import com.linkedin.thirdeye.datalayer.dto.DetectionConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
+import com.linkedin.thirdeye.datasource.DAORegistry;
+import java.util.HashMap;
+import java.util.Map;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class YamlDetectionConfigTranslatorTest {
+  private MetricConfigManager metricDAO;
+  private DAOTestBase testDAOProvider;
+
+  @Test
+  public void testGenerateDetectionConfig() {
+    Map<String, Object> properties = new HashMap<>();
+    properties.put("className", "test.linkedin.thirdeye");
+    YamlDetectionConfigTranslator translator = new MockYamlDetectionConfigTranslator();
+
+    Map<String, Object> yamlConfigs = new HashMap<>();
+    yamlConfigs.put("name", "testPipeline");
+    MetricConfigDTO metricConfigDTO = new MetricConfigDTO();
+    metricConfigDTO.setName("a_daily_metric");
+    metricConfigDTO.setDataset("a_test_dataset");
+    metricConfigDTO.setAlias("alias");
+    this.metricDAO.save(metricConfigDTO);
+
+    DetectionConfigDTO detectionConfigDTO = translator.generateDetectionConfig(yamlConfigs);
+    Assert.assertEquals(detectionConfigDTO.getName(), "testPipeline");
+    Assert.assertEquals(detectionConfigDTO.getCron(), "0 0 14 * * ? *");
+    Assert.assertEquals(detectionConfigDTO.getProperties().get("yamlConfigs"), yamlConfigs);
+  }
+
+  @BeforeMethod
+  public void setUp() {
+    this.testDAOProvider = DAOTestBase.getInstance();
+    DAORegistry daoRegistry = DAORegistry.getInstance();
+    this.metricDAO = daoRegistry.getMetricConfigDAO();
+  }
```

Comment:
add **@AfterMethod** method with **this.testDAOProvider.cleanup();**

Reply:
Thanks. Added after method.